server/integrationtests/user_test.go

```diff
@@ -0,0 +1,200 @@
+package integrationtests
+
+import (
+	"testing"
+
+	"github.com/mattermost/focalboard/server/api"
+	"github.com/mattermost/focalboard/server/utils"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	fakeUsername = "fakeUsername"
+	fakeEmail    = "mock@test.com"
+)
+
+func TestUserRegister(t *testing.T) {
+	th := SetupTestHelperWithoutToken().InitBasic()
+	defer th.TearDown()
+
+	// register
+	registerRequest := &api.RegisterRequest{
+		Username: fakeUsername,
+		Email:    fakeEmail,
+		Password: utils.CreateGUID(),
+	}
+	success, resp := th.Client.Register(registerRequest)
+	require.NoError(t, resp.Error)
+	require.True(t, success)
+
+	// register again will failed
+	success, resp = th.Client.Register(registerRequest)
+	require.Error(t, resp.Error)
+	require.False(t, success)
+}
+
+func TestUserLogin(t *testing.T) {
+	th := SetupTestHelperWithoutToken().InitBasic()
+	defer th.TearDown()
+
+	t.Run("with nonexist user", func(t *testing.T) {
+		loginRequest := &api.LoginRequest{
+			Type:     "normal",
+			Username: "nonexistuser",
+			Email:    "",
+			Password: utils.CreateGUID(),
+		}
+		data, resp := th.Client.Login(loginRequest)
+		require.Error(t, resp.Error)
+		require.Nil(t, data)
+	})
+
+	t.Run("with registered user", func(t *testing.T) {
+		password := utils.CreateGUID()
+		// register
+		registerRequest := &api.RegisterRequest{
+			Username: fakeUsername,
+			Email:    fakeEmail,
+			Password: password,
+		}
+		success, resp := th.Client.Register(registerRequest)
+		require.NoError(t, resp.Error)
+		require.True(t, success)
+
+		// login
+		loginRequest := &api.LoginRequest{
+			Type:     "normal",
+			Username: fakeUsername,
+			Email:    fakeEmail,
+			Password: password,
+		}
+		data, resp := th.Client.Login(loginRequest)
+		require.NoError(t, resp.Error)
+		require.NotNil(t, data)
+		require.NotNil(t, data.Token)
+	})
+}
+
+func TestGetUserMe(t *testing.T) {
+	th := SetupTestHelperWithoutToken().InitBasic()
+	defer th.TearDown()
+
+	t.Run("not login yet", func(t *testing.T) {
+		me, resp := th.Client.GetUserMe()
+		require.Error(t, resp.Error)
+		require.Nil(t, me)
+	})
+
+	t.Run("logged in", func(t *testing.T) {
+		// register
+		password := utils.CreateGUID()
+		registerRequest := &api.RegisterRequest{
+			Username: fakeUsername,
+			Email:    fakeEmail,
+			Password: password,
+		}
+		success, resp := th.Client.Register(registerRequest)
+		require.NoError(t, resp.Error)
+		require.True(t, success)
+		// login
+		loginRequest := &api.LoginRequest{
+			Type:     "normal",
+			Username: fakeUsername,
+			Email:    fakeEmail,
+			Password: password,
+		}
+		data, resp := th.Client.Login(loginRequest)
+		require.NoError(t, resp.Error)
+		require.NotNil(t, data)
+		require.NotNil(t, data.Token)
+
+		// get user me
+		me, resp := th.Client.GetUserMe()
+		require.NoError(t, resp.Error)
+		require.NotNil(t, me)
```

Comment:
If it's okay, could you just add a few checks to make sure the data in `me` matches the above data that you used to register the user (you are really getting back your own data and not someone elses) like you've done in other tests.

Reply:
Add some checks, done